libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -1000,6 +1017,19 @@ protected virtual Task<ResourceResponse> ProcessOutgoingActivityAsync(ITurnConte
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Logic to build an <see cref="AppCreentials"/> object to be used to acquire tokens
+        /// for this HttpClient.
+        /// </summary>
+        /// <param name="appId">The application id.</param>
+        /// <param name="oAuthScope">The optional OAuth scope.</param>
+        /// <returns>The app credentials to be used to acquire tokens.</returns>
+        protected virtual async Task<AppCredentials> BuildCredentialsAsync(string appId, string oAuthScope = null)
+        {
+            var appPassword = await CredentialProvider.GetAppPasswordAsync(appId).ConfigureAwait(false);
```

Comment:
Should this throw if the appId is Null/Empty?

Reply:
No, app id can be null or empty (it actually is in multiple test scnearios). Thanks for checking